library/std/src/sys/unix/thread.rs

```diff
@@ -505,24 +505,24 @@ pub mod guard {
     #[cfg(target_os = "macos")]
     unsafe fn get_stack_start() -> Option<*mut libc::c_void> {
         let th = libc::pthread_self();
-        let stackaddr =
-            libc::pthread_get_stackaddr_np(th) as usize - libc::pthread_get_stacksize_np(th);
-        Some(stackaddr as *mut libc::c_void)
+        let stackptr = libc::pthread_get_stackaddr_np(th);
+        Some(stackptr.map_addr(|addr| addr - libc::pthread_get_stacksize_np(th)))
     }
 
     #[cfg(target_os = "openbsd")]
     unsafe fn get_stack_start() -> Option<*mut libc::c_void> {
         let mut current_stack: libc::stack_t = crate::mem::zeroed();
         assert_eq!(libc::pthread_stackseg_np(libc::pthread_self(), &mut current_stack), 0);
 
+        let stack_ptr = current_stack.ss_sp;
```

Comment:
Semantically, shouldn't these be using `offset` instead of `with_addr` which is technically `wrapping_offset`? I realise the existing code isn't doing that but wondering if it should.

Reply:
Ah, that makes sense. Sorry, I'd been skimming and didn't notice these were void pointers, not u8 pointers.

Reply:
Yeah I made this exact """mistake""" the first time through this file and then got sad and went with this impl afterwards. 😿